GitTfs/Core/Changes/Git/RenameEdit.cs

```diff
@@ -6,24 +7,100 @@ public class RenameEdit : IGitChangedFile
         public string PathTo { get; private set; }
         public string NewSha { get; private set; }
+        public string OldSha { get; private set; }
         public string Score { get; private set; }
         public IGitRepository _repository { get; private set; }
 
         public RenameEdit(IGitRepository repository, GitChangeInfo changeInfo)
         {
             _repository = repository;
             NewSha = changeInfo.newSha;
+            OldSha = changeInfo.oldSha;
             Path = changeInfo.path;
             PathTo = changeInfo.pathTo;
             Score = changeInfo.score;
         }
 
         public void Apply(ITfsWorkspaceModifier workspace)
         {
-            workspace.Edit(Path);
-            workspace.Rename(Path, PathTo, Score);
-            var workspaceFile = workspace.GetLocalPath(PathTo);
-            _repository.CopyBlob(NewSha, workspaceFile);
+            if (Path.ToLower() != PathTo.ToLower())
+            {
+                var workspaceFile = workspace.GetLocalPath(PathTo);
+                workspace.Edit(Path);
+                workspace.Rename(Path, PathTo, Score);
+                _repository.CopyBlob(NewSha, workspaceFile);
+            }
+            else
+            {
+                //This change can not be mapped to TFS, since TFS is case insensitive
+                var workspaceFile = workspace.GetLocalPath(Path);
+                var newTestFile = workspaceFile + NewSha;
+                _repository.CopyBlob(NewSha, newTestFile);
+                var oldTestFile = workspaceFile + OldSha;
+                _repository.CopyBlob(OldSha, oldTestFile);
+                var changed = FileCompare(newTestFile, oldTestFile);
+                File.Delete(newTestFile);
+                File.Delete(oldTestFile);
+
+                //If the file was modified, record this, else ignore it
+                if (changed)
+                {
+                    workspace.Edit(Path);
+                    _repository.CopyBlob(NewSha, workspaceFile);
+                }
+            }
+        }
+
+        //From https://support.microsoft.com/en-us/help/320348/how-to-create-a-file-compare-function-in-visual-c
+        private bool FileCompare(string file1, string file2)
+        {
+            int file1byte;
+            int file2byte;
+            FileStream fs1;
+            FileStream fs2;
+
+            // Determine if the same file was referenced two times.
+            if (file1 == file2)
+            {
+                // Return true to indicate that the files are the same.
+                return true;
+            }
+
+            // Open the two files.
+            fs1 = new FileStream(file1, FileMode.Open);
+            fs2 = new FileStream(file2, FileMode.Open);
+
+            // Check the file sizes. If they are not the same, the files 
+            // are not the same.
+            if (fs1.Length != fs2.Length)
+            {
+                // Close the file
+                fs1.Close();
```

Comment:
`Close()` could be removed if using `using()`. That will make the code shorter...